sharding-core/sharding-core-common/src/main/java/org/apache/shardingsphere/core/strategy/encrypt/EncryptTable.java

```diff
@@ -153,7 +153,17 @@ public String getCipherColumn(final String logicColumn) {
      * @return optional of sharding encryptor
      */
     public Optional<String> findShardingEncryptor(final String logicColumn) {
-        return columns.containsKey(logicColumn) ? Optional.of(columns.get(logicColumn).getEncryptor()) : Optional.<String>absent();
+        String originLogicColumnName = getOriginLogicColumnName(logicColumn);
+        return columns.containsKey(originLogicColumnName) ? Optional.of(columns.get(getOriginLogicColumnName(logicColumn)).getEncryptor()) : Optional.<String>absent();
+    }
+
+    private String getOriginLogicColumnName(final String logicColumn) {
```

Comment:
It is recommended to use Guava package to shorten this code logic.

Reply:
Thanks for review, and furthermore, which api do you mean?